XiEditor/XiScrollView.swift

```diff
@@ -0,0 +1,27 @@
+// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+class XiScrollView: NSScrollView {
+
+    // NOTE: overriding scrollWheel: is necessary in order to disable responsiveScrolling
```

Comment:
I get where this is coming from; responsive scrolling is a difficult tradeoff, with some unpleasant interactions (one of the biggies is that it routinely asks for lines outside the visible view, in preparation for scrolling). Fortunately, I think that tough tradeoff is going away when we do GPU drawing. In the meantime, my slight preference is to keep responsive scrolling, because (a) it is less of a change, and (b) it keeps us honest in a way, it's more of a stress test on the line-fetch logic.

Reply:
My intention, currently, is to add support for responsive scrolling back in a subsequent PR; removing it here makes the scroll behaviour much easier to reason about. 

In my experiments with responsive scrolling, it really _isn't_ much of a stress test; pre-drawing only happens when the main thread is otherwise idle. It is much less taxing on the line-fetch logic than is grabbing the scrollbar and moving it rapidly.

It also doesn't seem like it offers much to improves performance; even on my slow computer (2013 macbook air) in a release build of this branch I cannot make the editor miss a draw call while scrolling in jquery, at 44pts.

I'll start to look into bringing it back, but my preference would be to try and play around with going to layer-based drawing first; it will be easier to get that working when our behaviour is more predictable.